Content.Server/PDA/Ringer/RingerSystem.cs

```diff
@@ -53,15 +55,24 @@ private void OnCurrencyInsert(Entity<RingerComponent> ent, ref CurrencyInsertAtt
     /// <summary>
     /// Handles the <see cref="GenerateUplinkCodeEvent"/> for generating an uplink code.
     /// </summary>
-    private void OnGenerateUplinkCode(Entity<RingerUplinkComponent> ent, ref GenerateUplinkCodeEvent ev)
+    private void OnGenerateUplinkCode(Entity<RingerAccessUplinkComponent> ent, ref GenerateUplinkCodeEvent ev)
     {
-        var code = GenerateRingtone();
-
-        // Set the code on the component
-        ent.Comp.Code = code;
+        ev.Code = null;
+        ent.Comp.Code = null;
 
-        // Return the code via the event
-        ev.Code = code;
+        // This will realistically only take one attempt... But you never know...
+        for (var i = 0; i < UplinkCodeGenerationAttempts; i++)
+        {
+            var code = GenerateRingtone();
+            if (!TryMatchRingtoneToStore(code, out _))
+            {
+                // Set the code on the component
+                ent.Comp.Code = code;
+                // Return the code via the event
+                ev.Code = code;
+                break;
+            }
+        }
```

Comment:
use NameIdentifierSystem to generate unique IDs. If you generate a 6-digit number you can just match each note to an integer (probably automatically since it's an enum). 

This is very much a hellish hack to drop in here.